common/events/events_io.cpp

```diff
@@ -123,6 +123,32 @@ void AsyncWriterFromWriter::Cancel() {
 	}
 }
 
+ReaderFromAsyncReader::ReaderFromAsyncReader(AsyncReaderFromEventLoopFunc func) {
+	reader_ = func(event_loop_);
+}
+
+mio::ExpectedSize ReaderFromAsyncReader::Read(
+	vector<uint8_t>::iterator start, vector<uint8_t>::iterator end) {
+	size_t read;
+	error::Error err;
+	err = reader_->AsyncRead(start, end, [this, &read, &err](size_t n, error::Error inner_err) {
+		read = n;
+		err = inner_err;
```

Comment:
This double-assignment to `err` is a bit confusing. I understand it does the right thing because we don't quite care if `AsyncRead()` itself failed or if it called the handler with an error, but I'd still use two separate variables for it.

Reply:
see my comment below about squashing the improvement for this